provisioning/device/src/TransportSettings/ProvisioningClientTransportSettings.cs

```diff
@@ -67,7 +68,9 @@ public abstract class ProvisioningClientTransportSettings
         /// If incorrectly implemented, your device may fail to connect to IoT hub and/or be open to security vulnerabilities.
         /// <para>
         /// This feature is only applicable for HTTP, MQTT over TCP, MQTT over web socket, AMQP
-        /// over TCP. AMQP web socket communication does not support this feature.
+        /// over TCP. AMQP web socket communication does not support this feature. For users who want
+        /// this support over AMQP websocket, you must instead provide a <see cref="ClientWebSocket"/>
+        /// instance with the desired callback set.
         /// </para>
         /// </remarks>
         public RemoteCertificateValidationCallback RemoteCertificateValidationCallback { get; set; } = DefaultRemoteCertificateValidation;
```

Comment:
What happens if this is set to null?

Reply:
Added the required null checks in the HTTP/MQTT/AMQP layers where they were missing